indexer-opts/src/lib.rs

```diff
@@ -0,0 +1,358 @@
+                    Ok(last_processed_block_height) => last_processed_block_height,
+                    Err(err) => {
+                        tracing::warn!(
+                            target: LOGGING_PREFIX,
+                            "Failed to fetch `last_processed_block_height` from meta data. Falling back to provided `start_block_height`\n{:#?}",
+                            err,
+                        );
+                        self.start_block_height.unwrap_or_else(|| panic!("`__meta` for INDEXER ID {} doesn't exist `start-from-block-height` must be provided", self.indexer_id))
+                    }
+                }
+            }
+        };
+
+        Ok(match self.chain_id {
+            ChainId::Mainnet => config_builder.mainnet(),
+            ChainId::Testnet => config_builder.testnet(),
+        }
+        .start_block_height(start_block_height)
+        .build()?)
+    }
+
+    // Creates AWS Credentials for NEAR Lake
+    fn lake_credentials(&'static self) -> aws_types::credentials::SharedCredentialsProvider {
+        let provider = aws_types::Credentials::new(
+            self.lake_aws_access_key.clone(),
+            self.lake_aws_secret_access_key.clone(),
+            None,
+            None,
+            &self.indexer_id,
+        );
+        aws_types::credentials::SharedCredentialsProvider::new(provider)
+    }
+
+    // Creates AWS Shared Config for NEAR Lake
+    fn lake_aws_sdk_config(&'static self) -> aws_types::sdk_config::SdkConfig {
+        aws_types::sdk_config::SdkConfig::builder()
+            .credentials_provider(self.lake_credentials())
+            .region(aws_types::region::Region::new("eu-central-1"))
+            .build()
+    }
+}
+
+impl FromStr for ChainId {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "mainnet" => Ok(Self::Mainnet),
+            "testnet" => Ok(Self::Testnet),
+            _ => Err(format!(
+                "Invalid CHAIN_ID: `{}`. Try `mainnet` or `testnet`",
+                s
+            )),
+        }
+    }
+}
+
+impl FromStr for StartMode {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "from-latest" => Ok(Self::FromLatest),
+            "from-interruption" => Ok(Self::FromInterruption),
+            _ => Err(format!(
+                "Invalid START_MODE: `{}`. Try `from-latest` or `from-interruption`",
+                s
+            )),
+        }
+    }
+}
+
+async fn final_block_height(rpc_url: &str) -> u64 {
+    let client = JsonRpcClient::connect(rpc_url);
+    let request = methods::block::RpcBlockRequest {
+        block_reference: BlockReference::Finality(Finality::Final),
+    };
+
+    let latest_block = client
+        .call(request)
+        .await
+        .unwrap_or_else(|_| panic!("Failed to fetch final block from RPC {}", rpc_url));
```

Comment:
`expect`?

Reply:
`clippy` insisted I should replace `expect` with `unwrap_or_else(|| panic!());` Who am I to argue with the `clippy`? :)